clang/lib/Sema/SemaChecking.cpp

```diff
@@ -6848,6 +6848,14 @@ CXXRecordMembersNamed(StringRef Name, Sema &S, QualType Ty) {
   return Results;
 }
 
+QualType Sema::getTypeMember(QualType Type, StringRef Name) {
+  auto Results = CXXRecordMembersNamed<TypeDecl>(Name, *this, Type);
+  assert(Results.size() <= 1);
+  if (Results.empty())
+    return {};
+  return Context.getTypeDeclType(*Results.begin());
+}
```

Comment:
This is bypassing access and ambiguity diagnostics, which doesn't seem correct given that a `common_type` specialization could have a private `type` member.